### PR TITLE
[2201.2.x] Fix NPE for field access expression on record json field

### LINKED ISSUE #37988: [Bug]: Compiler crashes for field access expression
### Description

Run the following
```ballerina
public function main() {
    
}

function foo() returns error? {
    json|error response = {};
    if (response is error) {
        return error(INTERNAL_SERVER_ERROR);
    }
    RudderErrorResponse|error errResponse = response.cloneWithType();
    if (errResponse is error) {
        return error(INTERNAL_SERVER_ERROR);
    }
    string existingComponentId = check errResponse.errors.additional_data.componentId;
}

public enum InternalError {
    INTERNAL_SERVER_ERROR = "Internal server error",
    CASTING_ERROR = "Type casting error",
    DEPLOYMENT_ERROR = "Error while deploy",
    PERSISTANCE_ERROR = "Database error"
}

public type RudderErrorResponse record {
    json errors;
};
```

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.1.1
2201.2.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #38328: [Bug]: NPE for field access expression on record json field
### Description

$title

### Steps to Reproduce

```
type FooRec1 record {
    json j;
};

public function main() returns error? {
    FooRec1 rec1 = {j: "1"};
    rec1 = {j: {k: {l: "1"}}};
    json|error val = rec1.j.k.l.p;
}
```
This is a run-time error `{ballerina}JSONOperationError` instead this result in NPE.

### Affected Version(s)

Branch 2201.2.x

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose

Fixes #38328
Fixes #37988

## Approach
This works properly in update 3 and master. So, cherry picked the changes from the master.

## Samples
```
type FooRec1 record {
    json j;
};

public function main() returns error? {
    FooRec1 rec1 = {j: "1"};
    rec1 = {j: {k: {l: "1"}}};
    json|error val = rec1.j.k.l.p;
}
```
This gave NPE in 2201.2.2. Now it is fixed

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -9458,9 +9460,14 @@ private void handleSafeNavigation(BLangAccessExpression accessExpr, BType type,
          */
 
         BLangExpression matchExpr = accessExpr.expr;
-        BType matchExprType = accessExpr.expr.getBType();
+        BLangSimpleVariableDef variableDef =
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:9463]: Shall we update the comment (https://github.com/ballerina-platform/ballerina-lang/pull/38419/files#diff-a5961a28254babde0aa44e4c1cb05126b5ee68494baccca1c76879dec267b8aeR9449-R9459) according to the new desugared code

Comment (prakanth97) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:9463]: Updated 8959c0339b0f19dbdde5ffbaadc300b8044f9353

**Comment by MaryamZi** [comment]
@prakanth97, was this change merged to other branches? The issues are still open.

**Comment by prakanth97** [comment]
> @prakanth97, was this change merged to other branches? The issues are still open.

we can close this issue https://github.com/ballerina-platform/ballerina-lang/issues/38328 since it is only appeared in 2.x

**Pull request merged**